app/src/main/java/com/github/se/travelpouch/ui/documents/DocumentPreview.kt

```diff
@@ -109,12 +133,30 @@ fun DocumentPreview(documentViewModel: DocumentViewModel, navigationActions: Nav
             actions = {
               IconButton(
                   onClick = {
-                    documentViewModel.deleteDocumentById(documentContainer.ref.id)
+                    val activitiesLinkedToDocument =
+                        activityViewModel.activities.value.filter {
+                          it.documentsNeeded.contains(documentContainer)
+                        }
+                    documentViewModel.deleteDocumentById(
+                        documentContainer, activitiesLinkedToDocument)
                     navigationActions.goBack()
                   },
                   modifier = Modifier.testTag("deleteButton")) {
                     Icon(imageVector = Icons.Default.Delete, contentDescription = "Delete Document")
                   }
+
+              IconButton(
+                  onClick = {
+                    if (activityViewModel.activities.value.isEmpty()) {
+                      Toast.makeText(context, "No Activities to available", Toast.LENGTH_LONG)
+                          .show()
```

Comment:
Could you please change the toast to "Cannot link image if there are no activities" ?

Reply:
Done